tests/virt_autotest/virt_utils.pm

```diff
@@ -625,4 +625,37 @@ sub perform_guest_restart {
     }
 }
 
+sub collect_host_and_guest_logs {
+    my ($guest_wanted, $host_extra_logs, $guest_extra_logs) = @_;
+
+    my $wait_script                = "1800";
+    my $logs_collector_script_name = "virt_logs_collector.sh";
+    my $logs_fetching_script_name  = "fetch_logs_from_guest.sh";
+    my $logs_collector_script_url  = data_url("virt_autotest/$logs_collector_script_name");
+    my $logs_fetching_script_url   = data_url("virt_autotest/$logs_fetching_script_name");
+
+    my $download_bash_script = "curl -s -o ~/$logs_collector_script_name $logs_collector_script_url";
+    script_output($download_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    save_screenshot;
+
+    my $execute_bash_script = "chmod +x ~/$logs_collector_script_name && ~/$logs_collector_script_name";
+    script_output("$execute_bash_script -l \"$host_extra_logs\" -g \"$guest_wanted\" -e \"$guest_extra_logs\"", $wait_script, type_command => 0, proceed_on_failure => 1);
+    save_screenshot;
+
+    $download_bash_script = "curl -s -o ~/$logs_fetching_script_name $logs_fetching_script_url";
+    script_output($download_bash_script, $wait_script, type_command => 0, proceed_on_failure => 0);
+    save_screenshot;
+
+    $execute_bash_script = "chmod +x ~/$logs_fetching_script_name && ~/$logs_fetching_script_name";
+    script_output("$execute_bash_script -g \"$guest_wanted\" -e \"$guest_extra_logs\"", $wait_script, type_command => 0, proceed_on_failure => 1);
+    save_screenshot;
```

Comment:
I think I have given comments in the other PR about wrapping download&execute bash script into a function and call it wherever needs such script execution from openqa data dir.

Reply:
You can not wrap everything all in one function. Executing scripts is very diverse, it is not just "chmod +x script; ./script". You end up passing long trailing options/arguments/parameters, I can not see much convenience it will bring.

Reply:
The correct thing to do is to wrap up scirpts that serve the same purpose into on function which includes their download and execution. The example is collect_host_and_guest_logs in virt_utils, I have already done that.

Reply:
> The correct thing to do is to wrap up scirpts that serve the same purpose into one function which includes their download and execution.

Exactly. There are 4~5 places with similar code that download script from openqa data directory, then chmod,then execute. Why can not these be put in a function? If you worry about too many different parameters to given to the downloaded script when executing, it can be quite easily be solved by combining these parameters into one.
It can look like this:
download_and_execute_script_from_openqa_data(script_name, execution_parameters){
  download
  chmod
  execute
}

 When you want to use it for the above last call with script_output("$execute_bash_script -g \"$guest_wanted\" -e \"$guest_extra_logs\"", $wait_script, type_command => 0, proceed_on_failure => 1), you can call download_and_execute_script_from_openqa_data($logs_fetching_script_name, "-g \"$guest_wanted\" -e \"$guest_extra_logs\"").

Right?